### ISSUE TITLE
A question about the code of Attribute Loss

### ISSUE THREAD

**Issue by znithx** [ErgastiAlex/MARS#9]
Hello，you mentioned that the Attribute Loss is computed with different weights based on how often they occur in the dataset.

![Image](https://github.com/user-attachments/assets/fe72e373-8c9c-4407-87d4-1676b64da142)

But in the code you posted, I didn't find this part about weight calculations，can you tell me where this code is ？
Here's the code for the Attribute loss calculation

        loss_attribute = 0
        count=0
        for attribute_mask, text_emb in zip(text2.attribute_masks, output_pos.last_hidden_state):
            max_attribute_value=torch.max(attribute_mask).to(torch.int32)
            
            # P-itm with attribute
            averaged_attribute=[]

            for i in range(1,max_attribute_value+1):
                mask=attribute_mask==i

                averaged_attribute.append(text_emb[mask].mean(0))
            
            if len(averaged_attribute)>0:
                vl_avg_output=self.itm_head(torch.stack(averaged_attribute))
                loss_attribute += F.cross_entropy(vl_avg_output, torch.ones(vl_avg_output.size(0), dtype=torch.long, device=image1.device))
                count += 1

        for attribute_mask, text_emb in zip(text_attribute_masks_neg, output_neg_cross.last_hidden_state):
            max_attribute_value=torch.max(attribute_mask).to(torch.int32)

            # P-itm with attribute
            averaged_attribute=[]

            for i in range(1,max_attribute_value+1):
                mask=attribute_mask==i

                averaged_attribute.append(text_emb[mask].mean(0))
            
            if len(averaged_attribute)>0:
                vl_avg_output=self.itm_head(torch.stack(averaged_attribute))
                loss_attribute += F.cross_entropy(vl_avg_output, torch.zeros(vl_avg_output.size(0), dtype=torch.long, device=image1.device))
                count += 1
        
        loss_attribute=loss_attribute/count

Thanks you！

**Comment by ErgastiAlex** [comment]
In the final version of the code the attribute loss is not weighted, since we observed minimal improvement only in the R1 and not in the other metrics, as shown by the ablation table

**Comment by znithx** [comment]
Thank you very much！
